### ISSUE TITLE
code wrong 

### ISSUE THREAD

**Issue by GYcumt** [ZhuoYulang/IF-MMIN#6]
encoder = getattr(self, 'encoder_' + str(i))
            decoder = getattr(self, 'decoder_' + str(i))
            x_in = x_in + shared # x_out
            latent = encoder(x_in)
            x_out = decoder(latent)
            latents.append(latent)

         博主你好，  x_in = x_in + shared # x_out  =这一行 改成 “  x_in = x_out + shared  # x_out ” 才对吧 我不太理解源代码的含义，可能是我理解错了，论文里面的公式和这个代码不对应呀

**Comment by ZhuoYulang** [comment]
您好！
非常感谢您指出这个问题。也很荣幸您能够仔细阅读我的代码。
这里确实是我的一个问题，我在同步GitHub文件的时候，这个文件传错了。这封邮件附上的是正确的文件。请您查收。
再次感谢您的指正。
祝您科研顺利~
---- Replied Message ----
***@***.***>Date9/4/2024 ***@***.***>***@***.***>Subject[ZhuoYulang/IF-MMIN] code wrong (Issue #6)
encoder = getattr(self, 'encoder_' + str(i))
decoder = getattr(self, 'decoder_' + str(i))
x_in = x_in + shared # x_out
latent = encoder(x_in)
x_out = decoder(latent)
latents.append(latent)
博主你好， x_in = x_in + shared # x_out =这一行 改成 “ x_in = x_out + shared # x_out ” 才对吧 我不太理解源代码的含义，可能是我理解错了，论文里面的公式和这个代码不对应呀

**Comment by GYcumt** [comment]
感谢您的回复，可是我并未收到您所说的邮件附带的代码，麻烦您检查一下或者可以的话发到我的邮箱13793785903@163.com
再次感谢，很棒的工作！
祝科研顺利！

**Comment by ZhuoYulang** [comment]
您好，非常感谢您对我工作的认可。
我重新添加了一次附件，麻烦您查看一下这次是否收到了呢？
如果还没有的话，可以从GItHub上进行查看。我已经更新了GitHub上的代码。
再次感谢~
顺颂时祺
---- Replied Message ----
***@***.***>Date9/4/2024 ***@***.***>***@***.***>,
***@***.***>SubjectRe: [ZhuoYulang/IF-MMIN] code wrong (Issue #6)
***@***.***
再次感谢，很棒的工作！
祝科研顺利！

**Comment by GYcumt** [comment]
感谢已经看到了！
但我还有一个问题，for i in range(self.n_blocks):
            encoder = getattr(self, 'encoder_' + str(i))
            decoder = getattr(self, 'decoder_' + str(i))
            x_in = x_in + shared # x_out
            latent = encoder(x_in)
            x_out = decoder(latent)
            latents.append(latent)
        latents = torch.cat(latents, dim=-1)
        return self.transition(x_in+x_out), latents
按照您的论文fig2所示，IF-IM模块输出的应该是最后一个自编码器的输出，但在您的代码’
 return self.transition(x_in+x_out),latents‘中 return的却是最后一个编码器的输入和输出的相加，我不明白是我理解错了还是另有其他的含义，希望您的解答！
再次感谢您的耐心回复！
祝一切顺利！

**Comment by ZhuoYulang** [comment]
再次感谢您能够仔细阅读我的代码。
您的理解并没有问题，这里是我们的一个失误。正确的代码逻辑应该是
"return self.transition(x_out), latents"
我们已经在GitHub中做出了修改。
再次感谢您的指正！
顺颂时祺~
---- Replied Message ----
***@***.***>Date9/4/2024 ***@***.***>***@***.***>,
***@***.***>SubjectRe: [ZhuoYulang/IF-MMIN] code wrong (Issue #6)
感谢已经看到了！
但我还有一个问题，for i in range(self.n_blocks):
encoder = getattr(self, 'encoder_' + str(i))
decoder = getattr(self, 'decoder_' + str(i))
x_in = x_in + shared # x_out
latent = encoder(x_in)
x_out = decoder(latent)
latents.append(latent)
latents = torch.cat(latents, dim=-1)
return self.transition(x_in+x_out), latents
按照您的论文fig2所示，IF-IM模块输出的应该是最后一个自编码器的输出，但在您的代码’
return self.transition(x_in+x_out),latents‘中 return的却是最后一个编码器的输入和输出的相加，我不明白是我理解错了还是另有其他的含义，希望您的解答！
再次感谢您的耐心回复！
祝一切顺利！

**Comment by leGao1** [comment]
博主，你好，我运行完代码，发现会创建一个shared_image文件里面，有十个1-10编码的文件夹和一个loss文件夹，然后每一个1-10为编码的文件夹中仍有三个小文件夹，但是运行完程序之后 没结果，然后我就检查了一下train_miss函数里面没有写LossRecorder类中的write_result_to_tsv方法，我想问一下这个要是写的画代码逻辑是什么，就是三个小文件夹中的loss  invariant predict 应该存储什么文件
